src/wms/src/main/java/org/geoserver/wms/featureinfo/HTMLFeatureInfoOutputFormat.java

```diff
@@ -68,7 +68,7 @@ public TemplateModel wrap(Object object) throws TemplateModelException {
         });
     }
 
-    GeoServerTemplateLoader templateLoader;
+    volatile GeoServerTemplateLoader templateLoader;
```

Comment:
Normally I see volatile being used along with a double checked locking pattern, but here it does not seem the case. Wasn't moving the loader init in the synchronized block enough?